python/mxnet/gluon/estimator/estimator.py

```diff
@@ -0,0 +1,314 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# coding: utf-8
+# pylint: disable=wildcard-import
+"""Gluon Estimator"""
+
+import warnings
+
+from .event_handler import LoggingHandler
+from ... import gluon, autograd
+from ...context import Context, cpu, gpu, num_gpus
+from ...io import DataIter
+from ...metric import EvalMetric, Loss
+import copy
+
+__all__ = ['Estimator']
+
+
+class Estimator(object):
```

Comment:
Wasn't this class included in this PR : https://github.com/apache/incubator-mxnet/pull/14346 ?

Reply:
Yes, I have made changes in it for validation support.

Reply:
I'd recommend either adding the changes for validation metrics as a commit to the existing PR #14346 , OR, wait until that PR gets merged, and add this as a singular commit so that the diff is incremental.

Reply:
@piyushghai I asked @abhinavs95 to open this PR so I can review it early. We can do a rebase and resolve the conflict once parent PR is merged.